### ISSUE TITLE
Unable to syncdb or migrate dumpdata

### ISSUE THREAD

**Issue by obfuscurity** [graphite-project/graphite-web#1425]
I'm updating [Synthesize](https://github.com/obfuscurity/synthesize) to be compatible with Graphite master but I'm running into some issues with the database migration. I typically perform a manual setup of the graphite.db and then `dumpdata` to create an `initial_data.json`, which I then use in an automated fashion (`--noinput`). This has worked well with all 0.9.x releases, but I'm hitting some issues with master.

I've tried a few different `dumpdata` settings, but `syncdb` or `migrate` always complains about missing tables. Here is one example.

```
vagrant@vagrant:/opt/graphite/webapp/graphite$ sudo PYTHONPATH=/opt/graphite/webapp django-admin.py migrate --noinput --settings=graphite.settings
Operations to perform:
  Synchronize unmigrated apps: account, render, staticfiles, whitelist, metrics, url_shortener, dashboard, composer, events, browser
  Apply all migrations: admin, contenttypes, tagging, auth, sessions
Synchronizing apps without migrations:
  Creating tables...
    Creating table account_profile
    Creating table account_variable
    Creating table account_view
    Creating table account_window
    Creating table account_mygraph
    Creating table dashboard_dashboard
    Creating table dashboard_template
    Creating table events_event
    Creating table url_shortener_link
    Running deferred SQL...
  Installing custom SQL...
Traceback (most recent call last):
  File "/usr/local/bin/django-admin.py", line 5, in <module>
    management.execute_from_command_line()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
...
...
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/sqlite3/base.py", line 318, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: Problem installing fixture '/opt/graphite/webapp/graphite/initial_data.json': Could not load contenttypes.ContentType(pk=1): no such table: django_content_type
```

I'm not what you'd consider a Django expert, so any feedback or suggestions would be greatly appreciated. The data is too large to paste in this report, so I've uploaded it to a gist [here](https://gist.github.com/obfuscurity/d46240979b3bebb22ccc).

/cc @brutasse @mleinart @deniszh

**Comment by obfuscurity** [comment]
Possibly related, we're also seeing [broken docs tests](https://travis-ci.org/graphite-project/graphite-web/jobs/95282396) in master related to the Django models, resulting in empty functions documentation. :cry: 

Running my tests locally:
```
...
...
reading sources... [ 95%] tools
reading sources... [ 97%] whisper
reading sources... [100%] who-is-using


Warning, treated as error:
/usr/local/src/graphite-web/docs/functions.rst:17: WARNING: autodoc: failed to import module u'graphite.render.functions'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/src/graphite-web/.tox/docs/local/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 385, in import_object
    __import__(self.modname)
  File "/usr/local/src/graphite-web/webapp/graphite/render/functions.py", line 26, in <module>
    from graphite.events import models
  File "/usr/local/src/graphite-web/webapp/graphite/events/models.py", line 4, in <module>
    from tagging.managers import ModelTaggedItemManager
  File "/usr/local/src/graphite-web/.tox/docs/local/lib/python2.7/site-packages/tagging/managers.py", line 5, in <module>
    from django.contrib.contenttypes.models import ContentType
  File "/usr/local/src/graphite-web/.tox/docs/local/lib/python2.7/site-packages/django/contrib/contenttypes/models.py", line 159, in <module>
    class ContentType(models.Model):
  File "/usr/local/src/graphite-web/.tox/docs/local/lib/python2.7/site-packages/django/db/models/base.py", line 94, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/usr/local/src/graphite-web/.tox/docs/local/lib/python2.7/site-packages/django/apps/registry.py", line 239, in get_containing_app_config
    self.check_apps_ready()
  File "/usr/local/src/graphite-web/.tox/docs/local/lib/python2.7/site-packages/django/apps/registry.py", line 124, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
AppRegistryNotReady: Apps aren't loaded yet.

ERROR: InvocationError: '/usr/local/src/graphite-web/.tox/docs/bin/sphinx-build -W -b html -d /usr/local/src/graphite-web/.tox/docs/tmp/doctrees . /usr/local/src/graphite-web/.tox/docs/tmp/html'
________________________________________ summary ________________________________________
ERROR:   docs: commands failed
```

**Comment by deniszh** [comment]
Will try to look...

**Comment by obfuscurity** [comment]
Locking in docs builds to Django==1.8 fixes that particular issue. I hate to do this but I think we need to limit our Django to `>=1.4,<1.9` for the time being.

**Comment by deniszh** [comment]
:+1: 
Looks like it's broken only in 1.9, which was released a week ago ...

**Comment by brutasse** [comment]
I'm fixing  the docs build, but the issue with `syncdb` comes from synthetize which provides a JSON fixture a bit too early it seems. I'd recommend doing `django-admin.py migrate` *then* `django-admin.py loaddata` with the path to the fixture. Using another name than `initial_data` prevents Django from picking it up during `migrate`.

**Comment by imbellish** [comment]
I've got a fork with some changes that allow Django 1.9 to ```migrate```, but there seems to be some major breaking changes with regards to django-tagging which is perhaps related. See: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=807454

**Comment by wawrzek** [comment]
https://bugs.launchpad.net/ubuntu/+source/graphite-web/+bug/1555621 . I found that 
    PYTHONPATH=/opt/graphite/webapp django-admin.py migrate --settings=graphite.settings --run-syncdb
works for me

**Comment by obfuscurity** [comment]
@wawrzek Yes indeed, that fixes it for me with Django 1.9. Thank you! 💯 😍 🍰 

@brutasse Your suggestion works for me with Django 1.8. I'm going to go ahead and use 1.9 going forward though. 😜

**Comment by vpereira** [comment]
@wawrzek  thank you :+1:
